File: chap1/SecondDegre.js
// Ces programmes sont sous licence CeCILL-B V1.
var a,b,c,delta;
a = ISNinputFloat();
b = ISNinputFloat();
c = ISNinputFloat();
delta = b * b - 4 * a * c;
if (delta<0){
	ISNprintln("Pas de solution");
}
else if (delta==0){
	ISNprintln("Une solution : ",-b/(2*a));
}
else {
	ISNprint ("Deux solutions : ");
	ISNprint ((- b - Math.sqrt(delta)) / (2*a));
	ISNprint(" et ");
	ISNprint ((- b + Math.sqrt(delta)) / (2*a));
}

File: chap1/SecondDegre.cpp
// Ces programmes sont sous licence CeCILL-B V1.
#include <iostream>
#include <cmath>

using namespace std;

int main()
{
   double a,b,c,delta;
   cin >> a;
   cin >> b;
   cin >> c;
   delta = b * b - 4 * a * c;
   if (delta < 0.0) 
   {
      cout << "Pas de solution" << endl;
   }
   else
   {
      if (delta == 0.0) 
      {
         cout << "Une solution : ";
         cout << - b / (2 * a) << endl;
      }
      else
      {
         cout << "Deux solutions : ";
         cout << (- b - sqrt(delta)) / (2 * a);
         cout << " et ";
         cout << (- b + sqrt(delta)) / (2 * a) << endl;
      }
   }
}
